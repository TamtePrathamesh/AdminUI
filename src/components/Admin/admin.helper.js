// export const getAdminList = () => {
//   return new Promise((resolve, reject) => {
//     fetch('privateURL')
//       .then((res) => res.json())
//       .then((res) => resolve(res))
//       .catch((res) => reject(res));
//   });
// };
export const getAdminList = async () => {
  try {
    let res = await fetch('privateURL');
    res = await res.json();
    return res;
  } catch (error) {
    console.error(error);
  }
};
