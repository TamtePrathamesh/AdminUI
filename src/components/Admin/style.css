
.adminContainer {
    padding: 20px;
    background-color: white;
}

.footerActions {
    display: flex;
    height: fit-content;
    align-items: center;
    justify-content: space-evenly;
    
}