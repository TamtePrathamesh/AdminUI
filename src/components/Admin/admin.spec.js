/* eslint-disable no-undef */
import React from 'react';
import { screen, render, waitFor, act } from '@testing-library/react';
import user from '@testing-library/user-event';
import * as helper from './admin.helper';
import '@testing-library/jest-dom';
import Admin from '.';

describe('Admin', () => {
  // jest.mock('./admin.helper', () => ({
  //   getAdminList: jest.fn().mockImplementation(() =>
  //     Promise.resolve([
  //       { id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: false },
  //       { id: '2', name: 'pqr', email: 'an@g.c', role: 'member', selection: false },
  //     ])
  //   ),
  // }));

  it('should match snapshot', () => {
    expect(render(<Admin />).container).toMatchSnapshot();
  });

  it('should display the data', async () => {
    render(<Admin />);

    await act(() => {
      waitFor(async () => await expect(screen.findByText('xyz')).toBeInTheDocument());
    });
    screen.debug();
  });
});
