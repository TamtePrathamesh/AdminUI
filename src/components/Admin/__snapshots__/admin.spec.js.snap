// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin should match snapshot 1`] = `
<div>
  <div
    class="adminContainer"
  >
    <input
      data-testid="search"
      id="search"
      placeholder="Search by name, email or role"
      type="text"
    />
    <div
      style="height: 15px;"
    />
    <div
      class="table-responsive-sm"
    >
      <table
        class="table"
      >
        <thead>
          <tr>
            <th>
              <input
                checked=""
                data-testid="selectAll"
                type="checkbox"
              />
            </th>
            <th
              scope="col"
            >
              Name
            </th>
            <th
              scope="col"
            >
              Email
            </th>
            <th
              scope="col"
            >
              Role
            </th>
            <th
              scope="col"
            >
              Actions
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      class="footerActions"
    >
      <div
        class="footer"
      >
        <button
          class="deleteSelected"
          data-testid="deleteSelected"
        >
          Delete Selected
        </button>
      </div>
      <div>
        <div
          class="pagination"
          data-testid="leading"
        >
          <div
            class="actionBtn"
          >
            <img
              alt="cant display"
              height="10px"
              src="leading-arrow.png"
              width="10px"
            />
          </div>
          <div
            class="actionBtn disableClick"
            data-testid="previous"
          >
            <img
              alt="cant display"
              height="10px"
              src="left-arrow.png"
              width="10px"
            />
          </div>
          <div
            class="contentOverflow"
          />
          <div
            class="actionBtn disableClick"
            data-testid="next"
          >
            <img
              alt="cant display"
              height="10px"
              src="right-arrow.png"
              width="10px"
            />
          </div>
          <div
            class="actionBtn"
            data-testid="trailing"
          >
            <img
              alt="cant display"
              height="10px"
              src="trailing-arrow.png"
              width="10px"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
